lib/services/cordova-project-plugins-service.ts

```diff
@@ -0,0 +1,682 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import {EOL} from "os";
+import * as util from "util";
+import * as pluginsDataLib from "../plugins-data";
+import Future = require("fibers/future");
+import * as helpers from "../common/helpers";
+import semver = require("semver");
+import * as cordovaPlugins from "./cordova-plugins";
+import * as validUrl from "valid-url";
+
+export class CordovaProjectPluginsService implements IPluginsService {
+	private static CORE_PLUGINS_PROPERTY_NAME = "CorePlugins";
+	private static CORDOVA_PLUGIN_VARIABLES_PROPERTY_NAME = "CordovaPluginVariables";
+	private static MESSAGES = ["Core Plugins", "Advanced Plugins", "Marketplace Plugins"];
```

Comment:
Not a merge-stopper, however I do prefer the name `HEADINGS` over the name `MESSAGES` - again, personal opinion